packages/core/admin/admin/src/content-manager/history/pages/History.tsx

```diff
@@ -1,20 +1,42 @@
 import * as React from 'react';
 
+import { createContext } from '@radix-ui/react-context';
 import { Flex } from '@strapi/design-system';
 import { LoadingIndicatorPage, useQueryParams } from '@strapi/helper-plugin';
+import { Contracts } from '@strapi/plugin-content-manager/_internal/shared';
 import { stringify } from 'qs';
 import { Helmet } from 'react-helmet';
 import { useIntl } from 'react-intl';
 import { Navigate, useNavigate, useParams } from 'react-router-dom';
 
 import { useContentTypeLayout } from '../../hooks/useLayouts';
 import { buildValidGetParams } from '../../utils/api';
+import { FormattedLayouts } from '../../utils/layouts';
 import { VersionDetails } from '../components/VersionDetails';
 import { VersionsList } from '../components/VersionsList';
 import { useGetHistoryVersionsQuery } from '../services/historyVersion';
 
 import type { UID } from '@strapi/types';
 
+/* -------------------------------------------------------------------------------------------------
+ * HistoryProvider
+ * -----------------------------------------------------------------------------------------------*/
+
+interface HistoryContextValue {
+  contentType: UID.ContentType;
+  id?: string; // null for single types
+  layout: FormattedLayouts;
+  selectedVersion: Contracts.HistoryVersions.HistoryVersionDataResponse;
+  versions: Contracts.HistoryVersions.GetHistoryVersions.Response;
+  page: number;
+}
+
+const [HistoryProvider, useHistoryContext] = createContext<HistoryContextValue>('HistoryPage');
```

Comment:
if you use the `createContext` from the https://github.com/strapi/strapi/blob/22f3123e946861ca798a41335ec80df6e68352e2/packages/core/admin/admin/src/components/Context.tsx#L5

you'll get a context selector included :)

Reply:
neat, thanks